### PR TITLE
Migrate to jupyter-ui-toolkit for Switch component

### LINKED ISSUE #124: Theme toggler extension overriding body font
## Description

Theme toggler extension imports [blueprint CSS](https://github.com/jupyterlab-contrib/jupyterlab-topbar/blob/712f910064831e304db2e32fc1ea0f683467c563/packages/theme-toggler-extension/src/index.tsx#L7) used for the  `Switch` component. As extensions are imported _after_ the core JupyterLab, Blueprint CSS is _potentially_ overriding JupyterLab core CSS properties. One change that is quite visible is body font family.

Here is a screenshot of font with vanilla JupyterLab:
jp_font

Here is a screenshot with theme toggler extension installed:
blueprint_font

Ideally, we should make the blueprint CSS scoped to theme toggler component.

## Reproduce

Install `jupyterlab-theme-toggler==1.0.0` and notice the change in font family

## Expected behavior

Extension should not alter JupyterLab core CSS properties

@jtpio I have been trying to scope it to component without much success. Do you have any leads? One solution is to redeclare the `body` CSS class with "correct" font-family in `base.css`:

```
body {
  font-family: var(--jp-ui-font-family);
}
```
or re-importing core CSS using `import '@jupyterlab/application/style/core.css';` should override the blueprint CSS and bring back JupyterLab default font family. If we can scope the blueprint CSS to the theme toggler component, that would be the ideal solution though!

### PR DESCRIPTION
**Pull request by mahendrapaipuri**
Fixes #124 

@jtpio I had some time to look at it. Let me know your thoughts!

### CODE REVIEW

**Comment by mahendrapaipuri** [comment]
Here is the binder link to test the PR -> https://mybinder.org/v2/gh/mahendrapaipuri/jupyterlab-topbar/use_jupyter_ui_components

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -85,6 +85,11 @@ jobs:
             py_cmd: python3
           - os: ubuntu
             py_cmd: python
+        # actions/setup-python do not support MacOS runners based on ARM for Python < 3.11
+        # https://github.com/actions/setup-python/issues/906
+        exclude:
```
Comment (jtpio) [inline .github/workflows/packaging.yml:90]: Wondering if we should remove the line `architecture: 'x64'` below in the `actions/setup-python`, and bump to `actions/setup-python@v5`?

Maybe we could then remove the extra `source ./venv/bin/activate`?

File: packages/theme-toggler-extension/src/index.tsx [hunk 2]
```diff
@@ -102,12 +45,11 @@ const extension: JupyterFrontEndPlugin<void> = {
     if (toolbarRegistry) {
       toolbarRegistry.addFactory('TopBar', 'theme-toggler', () => {
         const widget = ReactWidget.create(
-          <ThemeSwitch
-            themeManager={themeManager}
-            onChange={onChange}
-            innerLabel="light"
-            innerLabelChecked="dark"
-          />
+          <Switch onChange={onChange} checked={true}>
```
Comment (jtpio) [inline packages/theme-toggler-extension/src/index.tsx:48]: Should `checked` always be true?

Testing on Binder, the theme seems to be changed after a page refresh:

[theme-toggler.webm](https://github.com/user-attachments/assets/5e63be33-7aeb-4f35-bde7-64fa73bd1777)

**Review by jtpio — COMMENTED** [review]
Thanks @mahendrapaipuri for looking into this!

The new switch looks good. Maybe we should tweak the logic to take into account the current theme on page load?

**Comment by jtpio** [comment]
Also the theme name used to be "inside" the switch component:

image


Not sure there is a way to do this with the component from `jupyter-ui-toolkit`? And maybe we should remove the "Theme" label so it takes less space?

image

File: packages/theme-toggler-extension/src/index.tsx (hunk 2 above)
Comment (mahendrapaipuri) [reply packages/theme-toggler-extension/src/index.tsx]: Yes, you are correct. I over simplified the logic. Now I added the logic back to keep the state of the switch based on current theme.

**Comment by mahendrapaipuri** [comment]
> Also the theme name used to be "inside" the switch component:
> 
> image
> 
> Not sure there is a way to do this with the component from `jupyter-ui-toolkit`? And maybe we should remove the "Theme" label so it takes less space?
> 
> image

Yeah, there is no native way to embed labels inside the Switch component. Not sure if it is worth customizing the component. Maybe it makes more sense to address this issue in `jupyter-ui-toolkit`?

**Comment by mahendrapaipuri** [comment]
@jtpio Could you please take a look at it when you got some time? Cheers!

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**